staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -2269,3 +2261,299 @@ func TestLoadObjects(t *testing.T) {
+					}
+
+					if len(tokens) == 1 && tokens[0] == "namespaces" && method == "POST" {
+						data, err := io.ReadAll(req.Body)
+						if err != nil {
+							t.Fatalf("unexpected error: %v", err)
+						}
+
+						u := &unstructured.Unstructured{}
+						if err := runtime.DecodeInto(codec, data, u); err != nil {
+							t.Fatalf("unexpected error: %v", err)
+						}
+
+						name := u.GetName()
+						ns := u.GetNamespace()
+						gvr := schema.GroupVersionResource{Version: "v1", Resource: "namespaces"}
+
+						existing, err := fakeDynamicClient.Tracker().Get(gvr, ns, name)
+						if err != nil {
+							if apierrors.IsNotFound(err) {
+								existing = nil
+							} else {
+								t.Fatalf("error fetching object: %v", err)
+							}
+						}
+
+						uid := types.UID(fmt.Sprintf("%v", time.Now().UnixNano()))
+						u.SetUID(uid)
+
+						if existing == nil {
+							if err := fakeDynamicClient.Tracker().Create(gvr, u, ns); err != nil {
+								t.Fatalf("error creating object: %v", err)
+							}
+						} else {
+							if err := fakeDynamicClient.Tracker().Update(gvr, u, ns); err != nil {
+								t.Fatalf("error updating object: %v", err)
+							}
+						}
+
+						body := cmdtesting.ObjBody(codec, u)
+
+						return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: body}, nil
+					}
+
+					t.Fatalf("unexpected request: %v %v\n%#v", req.Method, req.URL, req)
+					return nil, nil
+				}),
+			}
+
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.ApplySet}, t, func(t *testing.T) {
+				manifests := []string{"manifest1", "manifest2"}
+				for _, manifest := range manifests {
+					t.Logf("applying manifest %v", manifest)
+
+					cmd := &cobra.Command{}
+					flags := NewApplyFlags(genericclioptions.NewTestIOStreamsDiscard())
+					flags.AddFlags(cmd)
+					cmd.Flags().Set("filename", filepath.Join(testdir, manifest+".yaml"))
+					cmd.Flags().Set("applyset", filepath.Base(testdir))
+					cmd.Flags().Set("prune", "true")
+					cmd.Flags().Set("validate", "false")
+
+					o, err := flags.ToOptions(tf, cmd, "kubectl", []string{})
+					if err != nil {
+						t.Fatalf("unexpected error creating apply options: %v", err)
+					}
+
+					err = o.Validate()
+					if err != nil {
+						t.Fatalf("unexpected error from validate: %v", err)
+					}
+
+					var unifiedOutput bytes.Buffer
+					o.Out = &unifiedOutput
+					o.ErrOut = &unifiedOutput
+
+					if err := o.Run(); err != nil {
+						t.Errorf("error running apply: %v", err)
+					}
+
+					got := unifiedOutput.String()
+
+					p := filepath.Join(testdir, manifest+"-expected-apply.txt")
+					wantBytes, err := os.ReadFile(p)
+					if err != nil {
+						t.Fatalf("error reading file %q: %v", p, err)
+					}
+					want := string(wantBytes)
+					if diff := cmp.Diff(want, got); diff != "" {
+						t.Errorf("apply output has unexpected diff (-want +got):\n%s", diff)
+					}
+				}
+			})
+		})
+	}
+}
+
+func TestApplyFail(t *testing.T) {
```

Comment:
same nit about the test name -- let's make it specific to our feature

Reply:
Done!